dref/views.py

```diff
@@ -45,6 +46,23 @@
 from dref.permissions import PublishDrefPermission
 
 
+def filter_dref_queryset_by_user_access(user, queryset):
+    if not user.is_superuser:
+        dref_admin_regions_id = [
+            codename.replace('dref_region_admin_', '')
+            for codename in Permission.objects.filter(
+                group__user=user,
+                codename__startswith='dref_region_admin_',
+            ).values_list('codename', flat=True)
+        ]
+        if len(dref_admin_regions_id):
+            return queryset.filter(
+                models.Q(created_by=user) | models.Q(country__region__in=dref_admin_regions_id)
+            ).distinct()
+        else:
+            return queryset.model.get_for(user)
```

Comment:
Let's move the is_superuser logic here as well.
```suggestion
    if user.is_superuser:
        return queryset
    # Check if regional admin
    dref_admin_regions_id = [
        codename.replace('dref_region_admin_', '')
        for codename in Permission.objects.filter(
            group__user=user,
            codename__startswith='dref_region_admin_',
        ).values_list('codename', flat=True)
    ]
    if len(dref_admin_regions_id):
        return queryset.filter(
            models.Q(created_by=user) | models.Q(country__region__in=dref_admin_regions_id)
        ).distinct()
    # Normal access
    return queryset.model.get_for(user)
```